models/models.js

```diff
@@ -20,4 +21,33 @@ exports.selectArticleById = (article_id) => {
       }
       return rows[0];
     });
-};
\ No newline at end of file
+};
+
+exports.selectArticles = () => {
+  const lookupObjPromise = db
+    .query(
+      `SELECT article_id, COUNT(article_id) as comment_count FROM comments GROUP BY article_id;`
+    )
+    .then(({ rows }) => {
+      const lookupObj = createLookupObject(rows, "article_id", "comment_count");
+      return lookupObj;
+    });
+  const articlesPromise = db
+    .query(`SELECT * FROM articles ORDER BY created_at DESC;`)
+    .then(({ rows }) => {
+      return rows;
+    });
+  return Promise.all([articlesPromise, lookupObjPromise]).then((values) => {
+    const articles = values[0];
+    const lookupObj = values[1];
+    for (let i = 0; i < articles.length; i++) {
+      if (lookupObj.hasOwnProperty(articles[i].article_id)) {
+        articles[i].comment_count = +lookupObj[articles[i].article_id];
+      } else {
+        articles[i].comment_count = 0;
+      }
+      delete articles[i].body;
+    }
+    return articles;
+  });
+};
```

Comment:
This does work, and it's great to see a fully tested util function being implemented. It also shows really good understanding of Promise.all() and accessing of nested object properties.

That being said, your two queries could be combined into one using a join, which would save quite a bit of hassle!

You can also specify which columns you want in your query so you don't have to cycle through and remove each body individually.